types/usercentrics-browser-ui/index.d.ts

```diff
@@ -0,0 +1,146 @@
+     * @see https://usercentrics.com/docs/web/features/api/control-functionality/#changelanguage
+     * @param language - Two character country code, e.g. "en" = set language to English
+     */
+    changeLanguage: (language: string) => Promise<void>;
+
+    /**
+     * Programmatic way to clear the CMP localStorage entries
+     * @see https://usercentrics.com/docs/web/features/api/control-functionality/#clearusersession
+     */
+    clearUserSession: () => Promise<void>;
+
+    /**
+     * Programmatic way to deny all consents
+     * @see https://usercentrics.com/docs/web/features/api/control-functionality/#denyallconsents
+     */
+    denyAllConsents: () => Promise<void>;
+
+    /**
+     * Programmatic way to get the currently selected language in the CMP
+     * @see https://usercentrics.com/docs/web/features/api/control-functionality/#getactivelanguage
+     */
+    getActiveLanguage: () => Promise<string>;
+
+    /**
+     * Retrieves all the Consent Details
+     * @see https://usercentrics.com/docs/web/features/api/control-functionality/#getconsentdetails
+     */
+    getConsentDetails: () => Promise<ConsentDetails>;
+
+    /**
+     * Programmatic way to get the Controller ID
+     * @see https://usercentrics.com/docs/web/features/api/control-functionality/#getcontrollerid
+     */
+    getControllerId: () => Promise<string>;
+
+    /**
+     * Programmatic way to check if consent is required (no consent given or resurface). Return value is only available after UI has initialized.
+     * @see https://usercentrics.com/docs/web/features/api/control-functionality/#isconsentrequired
+     */
+    isConsentRequired: () => Promise<boolean>;
+
+    /**
+     * Programmatic way to check if the app is initialized
+     * @see https://usercentrics.com/docs/web/features/api/control-functionality/#isinitialized
+     */
+    isInitialized: () => Promise<boolean>;
+
+    /**
+     * Saves the consents after being updated.
+     * @see https://usercentrics.com/docs/web/features/api/control-functionality/#saveconsents
+     */
+    saveConsents: () => Promise<void>;
+
+    /**
+     * Updates consents for whole categories of services
+     * @see https://usercentrics.com/docs/web/features/api/control-functionality/#updatecategoriesconsents
+     */
+    updateCategoriesConsents: (categoriesConsents: CategoriesConsents) => Promise<void>;
+
+    /**
+     * Updates consents for individual or multiple services
+     * @see https://usercentrics.com/docs/web/features/api/control-functionality/#updateservicesconsents
+     */
+    updateServicesConsents: (servicesConsents: ServicesConsents) => Promise<void>;
+
+    /**
+     * Programmatic way to update TCF consents
+     * @see https://usercentrics.com/docs/web/features/api/control-functionality/#updatetcfconsents
+     */
+    updateTcfConsents: (tcfConsents: TCFConsents) => Promise<void>;
+}
+
+// not documented on the Usercentrics website
+export interface ConsentInput {
```

Comment:
**question:** Where is this used then?

Reply:
Maybe this is it? [BaseTCFUserDecision](https://usercentrics.com/docs/web/implementation/ui/interfaces/#basetcfuserdecision)

Reply:
It's used in the parameters for `updateCategoriesConsents`, `updateServicesConsents` and `updateTcfConsents`. But you're right, the interface you linked should be used instead here. Thanks!

Reply:
Reused that interface in [`4b820b1`](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/71282/commits/4b820b19bda5343ba9b57a57c8762b8939d315ee).